corgi/core/migrations/0037_custom_indexes.py

```diff
@@ -0,0 +1,26 @@
+# custom django migrations for setting pg indexes
+# for django-mptt specific performance optimisations on query and sorting
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("core", "0036_productcomponentrelation_core_produc_product_c88392_idx"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            "CREATE INDEX core_componentnode_tree_parent_lft_idx ON public.core_componentnode USING btree(tree_id, parent_id, lft)"  # noqa: E501
```

Comment:
generally agree - though these in particular (right now) are something we probably want to isolate apart from the models ... as they are the result of pure performance analysis ... once we have run for a bit we can port back into the model.

Reply:
This doesn't actually work anyway since MPTT generates the field names dynamically, so Django can't create indexes on these fields before the app registry is fully loaded.

Reply:
Resolving this after some discussion on chat: these raw indexes touch fields that are dynamically generated by the models so the in-Django model indexes won't work (unfortunately). Keeping this as is.